pkg/aws/ec2/api/helper.go

```diff
@@ -76,28 +76,28 @@ func NewEC2APIHelper(ec2Wrapper EC2Wrapper, clusterName string) EC2APIHelper {
 type EC2APIHelper interface {
 	AssociateBranchToTrunk(trunkInterfaceId *string, branchInterfaceId *string, vlanId int) (*ec2.AssociateTrunkInterfaceOutput, error)
 	CreateNetworkInterface(description *string, subnetId *string, securityGroups []string, tags []*ec2.Tag,
-		secondaryPrivateIPCount int, interfaceType *string) (*ec2.NetworkInterface, error)
+		secondaryPrivateIPCount int, ipV4PrefixCount int, interfaceType *string) (*ec2.NetworkInterface, error)
 	DeleteNetworkInterface(interfaceId *string) error
 	GetSubnet(subnetId *string) (*ec2.Subnet, error)
 	GetBranchNetworkInterface(trunkID *string) ([]*ec2.NetworkInterface, error)
 	GetInstanceNetworkInterface(instanceId *string) ([]*ec2.InstanceNetworkInterface, error)
 	DescribeNetworkInterfaces(nwInterfaceIds []*string) ([]*ec2.NetworkInterface, error)
 	DescribeTrunkInterfaceAssociation(trunkInterfaceId *string) ([]*ec2.TrunkInterfaceAssociation, error)
 	CreateAndAttachNetworkInterface(instanceId *string, subnetId *string, securityGroups []string, tags []*ec2.Tag,
-		deviceIndex *int64, description *string, interfaceType *string, secondaryIPCount int) (*ec2.NetworkInterface, error)
+		deviceIndex *int64, description *string, interfaceType *string, secondaryIPCount int, ipV4PrefixCount int) (*ec2.NetworkInterface, error)
```

Comment:
Would that be possible to use something like 
```
type IPResourceCount struct {
  SecondaryIPCount
  IPv4PrefixCount
}
```
instead of two individual params for IP and PD?

Reply:
Thanks for the suggestion. I've updated the APIs to use the struct.